apps/src/codegen.js

```diff
@@ -111,10 +111,44 @@ exports.workspaceCode = function(blockly) {
 };
 
 //
+// Property access wrapped in try/catch. This is in an indepedendent function
+// so the JIT compiler can optimize the calling function.
+//
+
+function safeReadProperty(object, property) {
```

Comment:
This was discovered while marshaling certain Event objects that contained enumerable properties that caused an exception if you tried to access them.